reactivesocket-core/src/main/java/io/reactivesocket/ClientReactiveSocket.java

```diff
@@ -179,62 +181,98 @@ public ClientReactiveSocket start(Consumer<Lease> leaseConsumer) {
     }
 
     private Flux<Payload> handleStreamResponse(Flux<Payload> request, FrameType requestType) {
-        return started.thenMany(() -> {
-            int streamId = nextStreamId();
+        return started.thenMany(new Supplier<Publisher<Payload>>() {
+            final UnicastProcessor<Payload> receiver = UnicastProcessor.create();
+            final int streamId = streamIdSupplier.nextStreamId();
+            volatile MonoProcessor<Void> subscribedRequests;
+            boolean firstRequest = true;
+
+            boolean isValidToSendFrame() {
+                return contains(streamId) && connection.availability() > 0.0 && !receiver.isTerminated();
+            }
 
-            UnicastProcessor<Payload> receiver = UnicastProcessor.create();
+            void sendOneFrame(Frame frame) {
+                if (isValidToSendFrame()) {
```

Comment:
Generally it means the duplex connection isn't healthy and there would be an error sending it  - i.e. its probably closed.